cpp/run3ScoutingLooper.C

```diff
@@ -0,0 +1,587 @@
+  }
+};
+
+float MUON_MASS = 0.10566;
+
+bool isGlobalMuon(unsigned int type) {
+  return type & (1<<1);
+}
+
+bool isTrackerMuon(unsigned int type) {
+  return type & (1<<2);
+}
+
+bool isStandAloneMuon(unsigned int type) {
+  return type & (1<<3);
+}
+
+struct Muon {
+  std::vector<std::vector<int>> vtxIdxs;
+  std::vector<unsigned int> saHits, saMatchedStats;
+  std::vector<unsigned int> muHits, muChambs, muCSCDT, muMatch, muMatchedStats, muExpMatchedStats, muMatchedRPC;
+  std::vector<unsigned int> pixHits, stripHits;
+  std::vector<unsigned int> pixLayers, trkLayers;
+  std::vector<int> ch;
+  std::vector<float> pt, eta, phi;
+  std::vector<float> chi2ndof;
+  std::vector<float> ecalIso, hcalIso, trackIso;
+  std::vector<float> ecalRelIso, hcalRelIso, trackRelIso;
+  std::vector<float> dxy, dxye, dz, dze;
+  std::vector<float> dxysig, dzsig;
+  std::vector<float> PFIsoChg, PFIsoAll;
+  std::vector<float> PFRelIsoChg, PFRelIsoAll;
+  std::vector<float> mindr, maxdr;
+  std::vector<float> mindrPF, mindrJet;
+  std::vector<TLorentzVector> vec;
+  std::vector<TString> type; // types = ["G & T","G & !T","!G & T","SA & !G & !T","!SA & !G & !T"]
+
+  void clear() {
+  vtxIdxs.clear();
+  saHits.clear(); saMatchedStats.clear();
+  muHits.clear(); muChambs.clear(); muCSCDT.clear(); muMatch.clear(); muMatchedStats.clear(); muExpMatchedStats.clear(); muMatchedRPC.clear();
+  pixHits.clear(); stripHits.clear();
+  pixLayers.clear(); trkLayers.clear();
+  ch.clear();
+  pt.clear(); eta.clear(); phi.clear();
+  chi2ndof.clear();
+  ecalIso.clear(); hcalIso.clear(); trackIso.clear();
+  ecalRelIso.clear(); hcalRelIso.clear(); trackRelIso.clear();
+  dxy.clear(); dxye.clear(); dz.clear(); dze.clear();
+  dxysig.clear(); dzsig.clear();
+  PFIsoChg.clear(); PFIsoAll.clear();
+  PFRelIsoChg.clear(); PFRelIsoAll.clear();
+  mindr.clear(); maxdr.clear();
+  mindrPF.clear(); mindrJet.clear();
+  vec.clear();
+  type.clear();
+  }
+};
+
+
+std::vector<std::string> selL1Seeds = {"L1_DoubleMu_15_7",
+                                       "L1_DoubleMu4p5er2p0_SQ_OS_Mass_Min7",
+                                       "L1_DoubleMu4p5er2p0_SQ_OS_Mass_7to18",
+                                       "L1_DoubleMu4_SQ_OS_dR_Max1p2",
+                                       "L1_DoubleMu4p5_SQ_OS_dR_Max1p2"};
+
+
+void run3ScoutingLooper(std::vector<TString> inputFiles, TString year, TString process, const char* outdir="temp_data", TString label="") {
+  // Output folders and files
+  fs::create_directory(outdir);
+  fs::permissions(outdir,fs::perms::owner_all | fs::perms::group_read | fs::perms::group_exec | fs::perms::others_read | fs::perms::others_exec);
+  TFile* fout = new TFile(TString(outdir)+"/output_"+process+"_"+year+label+".root", "RECREATE");
+  TTree* tout = new TTree("tout","Run3ScoutingTree");
+
+  // Branch variables
+  float PV_x, PV_y, PV_z;
+  SV SVs;
+  int nMuonAssoc;
+  Muon Muons;
+
+  // Branch definition
```

Comment:
I would add run, luminosityBlock and event number already, since it may be useful at later stages